docs(weave): Add documentation for Koog framework (#5397)
wandb/weave: docs/docs/guides/integrations/koog.md

# Koog

[Koog](https://docs.koog.ai/) is a Kotlin-based framework for building single-run and complex workflow agents. Koog includes built-in OpenTelemetry (OTEL) support and can export traces directly to W&B Weave, giving you rich visibility into prompts, completions, tool calls, and end-to-end agent execution.

With the Weave exporter enabled, Koog forwards OpenTelemetry spans to your Weave project so you can debug, analyze performance, and iterate faster.

![Koog](imgs/koog.png)

## Prerequisites

Set the following environment variables before running your agent:

```bash
export WEAVE_API_KEY="<your-api-key>"
export WEAVE_ENTITY="<your-entity>"           # your W&B team/entity
export WEAVE_PROJECT_NAME="koog-tracing"      # any project name; created on first use
```

## Install Koog (Gradle)

Add Koog to your Kotlin project (Kotlin DSL shown):

```kotlin
dependencies {
    implementation("ai.koog:koog-agents:LATEST_VERSION")
}
```

> Learn more about installing Koog [here](https://docs.koog.ai/).

Reviewer: ```suggestion
See Koog's documentation for [installation information](https://docs.koog.ai/).
```

## Enable Weave export (OpenTelemetry)

> Install Koog’s OpenTelemetry feature and add the Weave exporter. Doing so uses Weave’s OpenTelemetry endpoint under the hood and maps Koog spans to Weave traces.

Reviewer: ```suggestion
Install Koog’s OpenTelemetry feature and add the Weave exporter. This maps Koog spans to Weave traces using Weave’s OpenTelemetry endpoint.
```
